CMakeLists.txt

```diff
@@ -1262,19 +1267,19 @@ set_target_properties(${CSOUNDLIB} PROPERTIES
 
 if(BUILD_STATIC_LIBRARY)
     add_library(${CSOUNDLIB_STATIC} STATIC ${libcsound_SRCS})
-    target_include_directories(${CSOUNDLIB_STATIC} PRIVATE ${libcsound_private_include_dirs})
-    target_include_directories(${CSOUNDLIB_STATIC} PUBLIC ${libcsound_public_include_dirs})
+    target_link_libraries(${CSOUNDLIB_STATIC} PRIVATE
+        $<BUILD_INTERFACE:libcsound_only>
+        $<BUILD_INTERFACE:libcsound_internals>
+        $<BUILD_INTERFACE:opcode_internals>
+    )
```

Comment:
Why is this `build_interface`? As far as I know, it's not common to have this with `target_link_libraries`.

Reply:
Alright, we need to test this to be sure. I don't fully get it but let's see